pypeit/reduce.py

```diff
@@ -733,11 +859,74 @@ def __init__(self, sciImg, spectrograph, tslits_dict, par, tilts, **kwargs):
             slit_spat_pos = edgetrace.slit_spat_pos(self.tslits_dict)
             self.order_vec = self.spectrograph.order_vec(slit_spat_pos)
 
+    def get_platescale(self, iord):
+        """
+        Return the plate scale for the given current echelle order
+        based on the order index
+
+        Args:
+            iord (int):
+                Echelle order index
+
+        Returns:
+            float:
+
+        """
+        # sobj first
```

Comment:
I'm not following what is happening here. If get_platescale uses sobjs it should passed as an argument since there are many sobjs in this module. I'm also confused as to why you need sobjs at all. You have an order number, then you are grabbing an object on that order only to use its order number. Please consult the original code.

Reply:
There is no original code as this is new for `skip_optimal`.

I might prefer we go back to the `if/else` block that I had.
Making this child methods is both confusing and a bit prone
to trouble as you are seeing.  Agreed?

Reply:
dealt with.